libshpool/src/daemon/server.rs

```diff
@@ -730,38 +724,21 @@ impl Server {
             info!("reaped child shell: {:?}", waitable_child);
         });
 
-        let has_clear_screen =
-            term_db.raw_string_cap(termini::StringCapability::ClearScreen).is_some();
-        let needs_default_term = !has_clear_screen || term.is_none();
-
-        // inject the prompt prefix, if any
-        info!("injecting prompt prefix");
-        let prompt_prefix =
-            self.config.get().prompt_prefix.clone().unwrap_or(String::from(DEFAULT_PROMPT_PREFIX));
-        if let Some(shell_basename) = shell_basename {
-            if !prompt_prefix.is_empty() {
-                if let Err(err) = prompt::inject_prefix(
-                    &mut fork,
-                    shell_basename,
-                    &prompt_prefix,
-                    &header.name,
-                    needs_default_term,
-                ) {
-                    warn!("issue injecting prefix: {:?}", err);
-                }
-            } else {
-                // issue a clear even if we don't have a prompt to inject for consistency
-                // and to simplify motd handling
-                let script = if needs_default_term {
-                    // If we don't have a $TERM value or we have some wacky $TERM value for which
-                    // there is no ClearScreen code, set TERM to xterm so that we won't get a
-                    // warning and will generate a code we can scan for.
-                    "TERM=xterm clear\n"
-                } else {
-                    "clear\n"
-                };
-                let mut pty_master = fork.is_parent().context("expected parent")?;
-                pty_master.write_all(script.as_bytes()).context("running initial clear")?;
+        // Inject the prompt prefix, if any. For custom commands, avoid doing this
+        // since we have no idea what the command is so the shell code probably won't
+        // work.
+        if header.cmd.is_none() {
```

Comment:
Oh custom commands are a not-so-commonly used feature where you can do `shpool attach -c '/my/custom -command'` and then shpool will launch that command instead of your shell. I don't really use it myself, but I think it would be pretty useful for making sure something like a webserver sticks around after you disconnect from a host (you can obviously do this with simple nohup commands, but I think shpool is somewhat nice in that you can reconnect to check in on the output).

Reply:
ah yeah makes sense, thanks for explaining. nice that this extends to cases like that